src/lib/db/store.ts

```diff
@@ -139,6 +140,20 @@ export function createDbStoreCreator(
       saveStorageDataList(liteStorage, newStorageMap!)
     }, [])
 
+    const addCloudLink = useCallback(async (id: string, cloudId: number) => {
+      let newStorageMap: ObjectMap<NoteStorage>
+      setStorageMap(prevStorageMap => {
+        newStorageMap = produce(prevStorageMap, draft => {
+          if (prevStorageMap[id] != null) {
+            draft[id]!.cloudStorageId = cloudId
+          }
+        })
+        return newStorageMap
+      })
+
+      saveStorageDataList(liteStorage, newStorageMap!)
```

Comment:
I think using `useEffect` should be better to save storage data list when it is changed.